debezium-embedded/src/main/java/io/debezium/embedded/AsyncEngineConfig.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.embedded;
+
+import io.debezium.config.Field;
+
+/**
+ * Configuration options specific to {@link AsyncEmbeddedEngine}.
+ *
+ * @author vjuranek
+ */
+public interface AsyncEngineConfig extends EmbeddedEngineConfig {
+
+    int AVAILABLE_CORES = Runtime.getRuntime().availableProcessors();
+
+    /**
+     * An optional field that specifies the maximum amount of time to wait for task lifecycle operation, i.e. for starting and stopping the task.
+     */
+    Field TASK_MANAGEMENT_TIMEOUT_MS = Field.create("task.management.timeout.ms")
+            .withDescription("Time to wait for task's lifecycle management operations (starting and stopping), given in milliseconds. "
+                    + "Defaults to 5 seconds (5000 ms).")
+            .withDefault(5_000L)
```

Comment:
Why do we make this assumption and why is this required to be configurable? Isn't the actual process of starting a task an asynchronous process anyway and it should also be deterministic, e.g. retries retry-able errors like configured and fails with a task-specific timeout? There can be a timeout but do we really need users to configure it ans/or just allow a quite high timeout instead, at least the timeout should be higher when looking at experiences with certain environments (like in RHOC there were sometimes issues with starting up a connector took 2-5 mins or more)?

Reply:
I'd definitely keep this configurable but it is really a bit too techincal tuning detail.
How about changing it into `internal` parameter and raise the default to 30 s?
Based on the feedback we might make it public.

Reply:
@rk3rn3r when you run any async operation, you should always set some timeout, unless there is very good reason for it. Otherwise you risk to get stuck waiting for the result forever. Sure, I can increase the default to whatever value you like. I'd make it internal only in case the default is quite high, as @rk3rn3r mentioned, starting time can vary quite a lot across the environments so my expectation is changing this value may happen to many users.

Reply:
Thx for the change. But I can't see this is an `internal.` field yet, is it?